src/inmanta/deploy/scheduler.py

```diff
@@ -593,6 +682,74 @@ async def is_agent_running(self, name: str) -> bool:
         """
         return name in self._workers and self._workers[name].is_running()
 
+    def create_recurring_repair_for_resource(self, resource: ResourceIdStr) -> ScheduledTask | None:
+        """
+        Schedule individual periodic repair for a given resource iff the repair timer is specified as
+        an int
+        """
+
+        async def _repair() -> None:
+            async with self._scheduler_lock:
+                if self._state.resource_state[resource].blocked is BlockedStatus.YES:  # Can't deploy
+                    return
+                to_deploy: set[ResourceIdStr] = {resource}
+
+                self._work.deploy_with_context(
+                    to_deploy,
+                    reason=f"Individual repair triggered for resource {resource} because last "
+                    f"repair happened more than {self._repair_timer}s ago.",
+                    priority=TaskPriority.INTERVAL_REPAIR,
+                    deploying=self._deploying_latest,
+                )
+
+        assert isinstance(self._repair_timer, int)
+        interval_schedule: IntervalSchedule = IntervalSchedule(
+            interval=float(self._repair_timer), initial_delay=self._repair_timer
+        )
```

Comment:
Does it make sense to scheduler more then one deploy? Don't we need single shot tasks?

(question, not suggestion)

Reply:
> Does it make sense to scheduler more then one deploy?

What do you mean here ?

Reply:
Currently, the interval timer keeps firing every X seconds . 

So let's say, every 10 s

timeline:
1. t=0, timer fires, resource goes in queue, 
2. t=10, timer fires, resource goes in queue again 
3. t=15, resources is de-queued, removes timer

So if the queue is full, we keep on re-queueing, but it is only the first one that has any relevance.